keras_cv/metrics/coco/mean_average_precision.py

```diff
@@ -0,0 +1,316 @@
+    `keras_cv.utils.bbox`.
+
+    Each image in a dataset may have a different number of bounding boxes,
+    both in the ground truth dataset and the prediction set.  In order to
+    account for this, you may either pass a `tf.RaggedTensor`, or pad Tensors
+    with `-1`s to indicate unused boxes.  A utility function to perform this
+    padding is available at `keras_cv_.utils.bbox.pad_bbox_batch_to_shape`.
+
+    ```python
+    coco_map = keras_cv.metrics.COCOMeanAveragePrecision(
+        max_detections=100,
+        class_ids=[1]
+    )
+
+    y_true = np.array([[[0, 0, 10, 10, 1], [20, 20, 10, 10, 1]]]).astype(np.float32)
+    y_pred = np.array([[[0, 0, 10, 10, 1, 1.0], [5, 5, 10, 10, 1, 0.9]]]).astype(
+        np.float32
+    )
+    coco_map.update_state(y_true, y_pred)
+    coco_map.result()
+    # 0.24752477
+    ```
+    """
+
+    def __init__(
+        self,
+        class_ids,
+        recall_thresholds=None,
+        iou_thresholds=None,
+        area_range=None,
+        max_detections=100,
+        num_buckets=10000,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        # Initialize parameter values
+        self.iou_thresholds = iou_thresholds or [x / 100.0 for x in range(50, 100, 5)]
+        self.area_range = area_range
+        self.max_detections = max_detections
+        self.class_ids = class_ids
+        self.recall_thresholds = recall_thresholds or [x / 100 for x in range(0, 101)]
+        self.num_buckets = num_buckets
+
+        self.num_iou_thresholds = len(self.iou_thresholds)
+        self.num_class_ids = len(self.class_ids)
+
+        self.ground_truths = self.add_weight(
+            "ground_truths",
+            shape=(self.num_class_ids,),
+            dtype=tf.int32,
+            initializer="zeros",
+        )
+        self.true_positive_buckets = self.add_weight(
+            "true_positive_buckets",
+            shape=(
+                self.num_class_ids,
+                self.num_iou_thresholds,
+                self.num_buckets,
+            ),
+            dtype=tf.int32,
+            initializer="zeros",
+        )
+        self.false_positive_buckets = self.add_weight(
+            "false_positive_buckets",
+            shape=(
+                self.num_class_ids,
+                self.num_iou_thresholds,
+                self.num_buckets,
+            ),
+            dtype=tf.int32,
+            initializer="zeros",
+        )
+
+    def reset_state(self):
+        self.true_positive_buckets.assign(tf.zeros_like(self.true_positive_buckets))
+        self.false_positive_buckets.assign(tf.zeros_like(self.false_positive_buckets))
+        self.ground_truths.assign(tf.zeros_like(self.ground_truths))
+
+    @tf.function()
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        num_images = tf.shape(y_true)[0]
+
+        if sample_weight is not None:
+            raise ValueError("Received unsupported `sample_weight` to `update_state`")
```

Comment:
This makes it sound like the sample_weight format is just invalid -- but in fact there are no valid values. Perhaps "{self.__class__.__name__} does not support `sample_weight`"